pkg/universe.dagger.io/go/configure.cue

```diff
@@ -0,0 +1,102 @@
+package go
+
+import (
+	"dagger.io/dagger"
+	"dagger.io/dagger/core"
+	"universe.dagger.io/docker"
+)
+
+// Default values used for #Configure.
+#DefaultGoEnvs: {
+	GOROOT: "/usr/local/go"
+	GOPATH: "/go"
+}
+
+// Configure installs go into the given input image.
+// It sets GOROOT, GOPATH, and adds the relevent go bin paths to PATH.
+#Configure: {
+	input: docker.#Image
+	// path to install go to in `input` and set the GOROOT env var to
+	// In most cases you want to use the default value provided here.
+	goroot: string | *#DefaultGoEnvs.GOROOT
+	// path for GOPATH to be set to
+	// In most cases you want to use the default value provided here.
+	gopath: string | *#DefaultGoEnvs.GOPATH
+
+	// path to go root in the specified filesystem (be it an image ref or a dagger.#FS)
+	source: string | *#DefaultGoEnvs.GOROOT
+
+	{
+		// filesystem with go in it.
+		// This will be copied into the image at the path specified by `goroot`
+		// if `source` is specified just the path to the source will be copied.
+		contents: dagger.#FS
+		_cfg:     _#Configure & {
+			"input":    input
+			"source":   source
+			"goroot":   goroot
+			"gopath":   gopath
+			"contents": contents
+		}
+		output: _cfg.output
+	} | {
+		ref: core.#Ref
+
+		_pull: docker.#Pull & {
+			source: ref
+		}
+		_cfg: _#Configure & {
+			"input":  input
+			"source": source
+			"goroot": goroot
+			"gopath": gopath
+			contents: _pull.output.rootfs
+		}
+		output: _cfg.output
+	}
+}
```

Comment:
I **think** this conjunction can be expressed something like this: 

```suggestion
    _cfg: _#Configure & {
    	"input":    input
	"source":   source
	"goroot":   goroot
	"gopath":   gopath
    }
	{
		// filesystem with go in it.
		// This will be copied into the image at the path specified by `goroot`
		// if `source` is specified just the path to the source will be copied.
		contents: dagger.#FS
		_cfg:     _#Configure & {
			"contents": contents
		}

	} | {
		ref: core.#Ref
		_pull: docker.#Pull & {
			source: ref
		}
		_cfg: _#Configure & {
			contents: _pull.output.rootfs
		}
	}
	output: _cfg.output
}
```

Reply:
Oh nice!

Reply:
Made this change.